app/rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationRevisionHandler.java

```diff
@@ -0,0 +1,208 @@
+                .build();
+        }
+
+        return integrationRevision;
+    }
+
+
+    @Override
+    public ListResult<IntegrationRevision> list(UriInfo uriInfo) {
+        return getDataManager().fetchAll(
+            IntegrationRevision.class,
+            new IntegrationRevisionHandler.DeletedFilter(),
+            new ReflectiveSorter<>(IntegrationRevision.class, new SortOptionsFromQueryParams(uriInfo)),
+            new PaginationFilter<>(new PaginationOptionsFromQueryParams(uriInfo))
+        );
+    }
+
+    @Override
+    public IntegrationRevision create(@Context SecurityContext sec, @ConvertGroup(from = Default.class, to = AllValidations.class) final IntegrationRevision integrationRevision) {
+        Date rightNow = new Date();
+
+        IntegrationRevision encryptedIntegrationRevision = encryptionSupport.encrypt(integrationRevision);
+
+        IntegrationRevision updatedIntegrationRevision = new IntegrationRevision.Builder()
+            .createFrom(encryptedIntegrationRevision)
+            .lastUpdated(rightNow)
+            .createdDate(rightNow)
+            .currentState(determineCurrentState(encryptedIntegrationRevision))
+            .stepsDone(new ArrayList<>())
+            .build();
+
+        return Creator.super.create(sec, updatedIntegrationRevision);
+    }
+
+
+    @Path("/{id}/{version}")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public void update(@NotNull @PathParam("id") @ApiParam(required = true) String id, @NotNull @PathParam("version") @ApiParam(required = true) String version, @ConvertGroup(from = Default.class, to = AllValidations.class) IntegrationRevision integrationRevision) {
+        String compositeId = id +"#" + version;
+        IntegrationRevision existing = Getter.super.get(compositeId);
+
+        IntegrationRevisionState currentState = determineCurrentState(integrationRevision);
+
+        IntegrationRevision currentRevision = new IntegrationRevision.Builder().createFrom(existing)
+            .currentState(currentState)
+            //TODO: We possibly need to remove that as it should have no effect.
+            .targetState(existing.getTargetState() != null ? existing.getTargetState() : IntegrationRevisionState.Pending)
+            .build();
+
+        IntegrationRevision updatedIntegration = new IntegrationRevision.Builder()
+            .createFrom(encryptionSupport.encrypt(integrationRevision))
+            .lastUpdated(new Date())
+            .createdDate(currentRevision.getCreatedDate())
+            .currentState(currentState)
+            .stepsDone(new ArrayList<>())
+            .build();
+
+        Updater.super.update(compositeId, updatedIntegration);
+    }
+
+
+
+    @Path("/{id}/{version}")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public void delete(@NotNull @PathParam("id") @ApiParam(required = true) String id, @NotNull @PathParam("version") @ApiParam(required = true) String version) {
```

Comment:
We should never allow deleting deployment history.  This API should be blocked.

Reply:
And how would we express that we want a revision/deployment to be `Undeployed`, given that we've already agree that we won't accept updates via rest?

Reply:
There should be an API for that.  But it would not be to update the whole deployment model.
perhaps it's best with a POST /api/v1/integrations-status/undeploy